src/controllers/client.js

```diff
@@ -72,56 +80,89 @@ async function assignEchisClientUPI(req, res) {
-    const mismatchedFields = getMismatchedClientFields(res.locals.echisClient, res.locals.crClient);
+    const mismatchedFields = getMismatchedClientFields(
+      res.locals.echisClient,
+      res.locals.crClient
+    );
     // if fields do not match, create report on echis for follow up and return error
     if (Object.keys(mismatchedFields).length > 0) {
       try {
         await createClientDetailsMismatchReport(
-          { instance: CHT.url, user: CHT.username, password: CHT.password },
-          res.locals.echisClient._id, Object.keys(mismatchedFields)
+          instanceObject,
+          res.locals.echisClient._id,
+          Object.keys(mismatchedFields)
         );
         res.status(200).send({
           data: {
             msg: CLIENT_DETAILS_MISMATCH,
-            fields: mismatchedFields
-          }
-        })
+            fields: mismatchedFields,
+          },
+        });
       } catch (err) {
-        const errString = `could not create details mismatch report on ${CHT.url}: ${err.message}`
-        logger.error(errString)
-        res.status(400).send({ error: errString })
+        const errString = `could not create details mismatch report on ${
+          getCHTValuesFromEnv(instance).url
+        }: ${err.message}`;
+        logger.error(errString);
+        res.status(400).send({ error: errString });
       }
-    } else { // else update contact doc with UPI
+    } else {
+      // else update contact doc with UPI
       try {
-        await updateDoc({ instance: CHT.url, user: CHT.username, password: CHT.password }, res.locals.echisClient._id, { upi: res.locals.crClient.clientNumber });
-        success(200, res.locals.echisClient._id, res.locals.crClient.clientNumber);
+        await updateDoc(instanceObject, res.locals.echisClient._id, {
+          upi: res.locals.crClient.clientNumber,
+        });
+        success(
+          200,
+          res.locals.echisClient._id,
+          res.locals.crClient.clientNumber
+        );
       } catch (err) {
-        const errString = `could not update doc ${res.locals.echisClient._id} on ${CHT.url}: ${err.message}`
-        logger.error(errString)
-        res.status(400).send({ error: errString })
+        const errString = `could not update doc ${
+          res.locals.echisClient._id
+        } on ${getCHTValuesFromEnv(instance).url}: ${err.message}`;
+        logger.error(errString);
+        res.status(400).send({ error: errString });
       }
     }
-    return // nothing more to do here
+    return; // nothing more to do here
   }
   // echis client did not exist in the registry, lets create them
   try {
-    const clientNumber = await createCRClient({ instance: CHT.url, user: CHT.username, password: CHT.password }, res.locals.echisClient);
-    success(201, res.locals.echisClient._id, clientNumber)
+    const clientNumber = await createCRClient(
+      instanceObject,
+      res.locals.echisClient
+    ).catch((error) => {
```

Comment:
do not catch the error here, we have a try catch for that. Build the error message in the catch block